src/joint.jl

```diff
@@ -1,3 +1,22 @@
+struct Bounds{T}
+    lower::T
+    upper::T
+end
+
+struct JointBounds{T}
+    position::Bounds{Vector{T}}
+    velocity::Bounds{Vector{T}}
+    effort::Bounds{Vector{T}}
+end
```

Comment:
Is having a separate struct for bounds overkill? What do you about just having `position_bounds`, `velocity_bounds`, and `effort_bounds` fields in `Joint`?

Reply:
I thought about that, but I generally feel like a bunch of variables with multi-part names is a sign that there's actually a hierarchy that isn't being represented correctly. This way requires exactly the same number of characters to access and also means that the bounds are something we can talk about collectively, in case that's useful.

Reply:
I don't really agree with that first argument in the general sense, but since the (implied) API is to use the accessor functions, we're free to change it later and so this isn't blocking.

Reply:
Ok, I've removed the JointBounds type